File: java/cars-assemble/CarsAssemble.java
public class CarsAssemble {

    static int BASE_RATE = 221; // Base rate at lowest speed.

    public double productionRatePerHour(int speed) {

        double success_rate = 0.0; // Success rate for the given speed
        switch (speed) {

            case 1:
            case 2:
            case 3:
            case 4:
                success_rate = 1.0;
                break;
            case 5:
            case 6:
            case 7:
            case 8:
                success_rate = 0.9;
                break;
            case 9:
                success_rate = 0.8;
                break;
            case 10:
                success_rate = 0.77;
                break;
        }

        return BASE_RATE * speed * success_rate;
    }

    public int workingItemsPerMinute(int speed) {

        return (int) (productionRatePerHour(speed) / 60); // 1 60th of the hourly rate cast to int.
    }
}

File: csharp/cars-assemble/CarsAssemble.cs
using System;

static class AssemblyLine {
    const int BASE_RATE = 221;
    const double success1 = 1;
    const double success2 = 0.9;
    const double success3 = 0.8;
    const double success4 = 0.77;

    public static double ProductionRatePerHour(int speed) {

        double result = (BASE_RATE * speed);
        if (speed >= 1 && speed <= 4) {
            result *= success1;
        } else if (speed >= 5 && speed <= 8) {
            result *= success2;
        } else if (speed == 9) {
            result *= success3;
        } else if (speed == 10) {
            result *= success4;
        }

        return result;
    }

    public static int WorkingItemsPerMinute(int speed) {

        double hourly = ProductionRatePerHour(speed);
        int per_minute = (int)(hourly / 60);
        return per_minute;
    }
}
